app/javascript/packs/feedPreviewCards.jsx

```diff
@@ -0,0 +1,109 @@
+  if (fetched) {
+    return;
+  }
+  metadataPlaceholder.dataset.fetched = 'true';
+
+  const previouslyFetchedAuthorMetadata = cachedAuthorMetadata[authorId];
+
+  if (previouslyFetchedAuthorMetadata) {
+    renderMetadata(previouslyFetchedAuthorMetadata, metadataPlaceholder);
+  } else {
+    const response = await request(`/profile_preview_cards/${authorId}`);
+    const authorMetadata = await response.json();
+
+    cachedAuthorMetadata[authorId] = authorMetadata;
+    renderMetadata(authorMetadata, metadataPlaceholder);
+  }
+}
+
+function renderMetadata(metadata, placeholder) {
+  const container = placeholder.parentElement;
+
+  render(<MemoizedUserMetadata {...metadata} />, container, placeholder);
+
+  container
+    .closest('.profile-preview-card__content')
+    .style.setProperty('--card-color', metadata.card_color);
+}
+
+function checkForPreviewCardDetails(event) {
+  const { target } = event;
+
+  if (target.classList.contains('profile-preview-card__trigger')) {
+    const metadataPlaceholder = target.parentElement.getElementsByClassName(
+      'author-preview-metadata-container',
+    )[0];
+
+    if (metadataPlaceholder) {
+      // User is within one of the story cards - and the metadata has not been fetched yet
+      populateMissingMetadata(metadataPlaceholder);
+    }
+  }
+}
+
+function listenForHoveredOrFocusedStoryCards() {
+  const mainContent = document.getElementById('main-content');
+
+  mainContent.addEventListener('mouseover', checkForPreviewCardDetails);
+  mainContent.addEventListener('focusin', checkForPreviewCardDetails);
+}
+
+function initializeFeedPreviewCards() {
+  // Select all preview card triggers that haven't already been initialized
+  const allPreviewCardTriggers = document.querySelectorAll(
+    'button[id^=story-author-preview-trigger]:not([data-initialized])',
+  );
+
+  for (const previewTrigger of allPreviewCardTriggers) {
+    const dropdownContentId = previewTrigger.getAttribute('aria-controls');
+    const dropdownElement = document.getElementById(dropdownContentId);
+
+    if (dropdownElement) {
+      initializeDropdown({
+        triggerElementId: previewTrigger.id,
+        dropdownContentId,
+        onOpen: () => dropdownElement?.classList.add('showing'),
+        onClose: () => dropdownElement?.classList.remove('showing'),
+      });
+
+      previewTrigger.dataset.initialized = true;
+    }
+  }
+}
+
+const observer = new MutationObserver((mutationsList) => {
+  mutationsList.forEach((mutation) => {
+    if (mutation.type === 'childList') {
+      initializeFeedPreviewCards();
+    }
+  });
+});
+
+if (document.getElementById('index-container')) {
+  observer.observe(document.getElementById('index-container'), {
+    childList: true,
+    subtree: true,
+  });
+}
+
+InstantClick.on('change', () => {
+  observer.disconnect();
+});
+
+window.addEventListener('beforeunload', () => {
+  observer.disconnect();
+});
+
+initializeFeedPreviewCards();
+listenForHoveredOrFocusedStoryCards();
```

Comment:
Refactor (blocking): Use a [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports) since these preview cards are only for feed items that appear with infinite scrolling.

```suggestion
import('./someFile').then(
  ({ initializeFeedPreviewCards, listenForHoveredOrFocusedStoryCards }) => {
    initializeFeedPreviewCards();
    listenForHoveredOrFocusedStoryCards();
  },
);

```

Reply:
They're also needed for non-infinite-scroll feed preview cards that are present from the get-go. It's the mutation observer further up that picks up on newly inserted/paged in feed cards

Reply:
If they're used for non-infinite scrolling as well would it still be fine as a dynamic import? I'm on mobile ATM, but I imagine yes?

Reply:
I guess they could, but I'm not sure I understand what the benefit would be, since this pack is already loaded with `defer: true` and moving those functions to a separate file wouldn't leave anything for `feedPreviewCards` to do other than import them in - maybe I'm missing something here 🙃

Reply:
It is deferred, but my point was more about loading things dynamically that aren't required immediately for the page to load/work. It's true, the pack file would be doing pretty much the dynamic import and calling those functions, but that's OK. Typically, pack files, which are [webpack entry points](https://webpack.js.org/concepts/entry-points/), shouldn't have too much going on in them. Having said that, we have pack files with a bunch of stuff going on in them, lol.

Reply:
Yeah maybe that's my confusion 😂  I can split it out

Reply:
done 🚀  I'll keep this in mind for new packs too 😄